src/main/java/uk/nhs/hee/tis/trainee/forms/service/LtftService.java

```diff
@@ -89,4 +92,75 @@ public long getAdminLtftCount(Set<LifecycleState> states) {
 
     return ltftFormRepository.countByStatusIn(states);
   }
+
+  /**
+   * Get the 'id' LTFT form if it belongs to the current user.
+   *
+   * @return The LTFT form, or optional empty if not found or does not belong to user.
+   */
+  public Optional<LtftFormDto> getLtftForm(UUID formId) {
+    String traineeId = traineeIdentity.getTraineeId();
+    log.info("Getting LTFT form {} for trainee [{}]", formId, traineeId);
+
+    Optional<LtftForm> form = ltftFormRepository.findByTraineeIdAndId(traineeId, formId);
+    if (form.isPresent()) {
+      log.info("Found form {} for trainee [{}]", formId, traineeId);
+      return Optional.of(mapper.toDto(form.get()));
+    } else {
+      log.info("Did not find form {} for trainee [{}]", formId, traineeId);
+      return Optional.empty();
+    }
+  }
+
+  /**
+   * Save the dto as a new LTFT form.
+   *
+   * @param dto The LTFT DTO to save.
+   * @return The saved form DTO.
+   */
+  public Optional<LtftFormDto> saveLtftForm(LtftFormDto dto) {
+    String traineeId = traineeIdentity.getTraineeId();
+    log.info("Saving LTFT form for trainee [{}]: {}", traineeId, dto);
+    LtftForm form = mapper.toEntity(dto);
+    if (!form.getTraineeId().equals(traineeId)) {
+      log.warn("Could not save form since it does belong to the logged-in trainee {}: {}",
+          traineeId, dto);
+      return Optional.empty();
+    }
+    form.setId(UUID.randomUUID());
```

Comment:
manually set ID because mongo can't autogenerate UUIDs. There must be a better way of doing this. It also necessitates custom determination of what is a new entity, since we are force-populating the id before saving.